v6_carrier_py/master.py

```diff
@@ -89,3 +92,15 @@ def column_names(client: ClientContainerProtocol, data, *args, exclude_orgs=(),
 
     # return all the messages from the nodes
     return column_set
+
+
+def correlation_matrix(client: ClientContainerProtocol, data, keys=None, *args, **kwargs):
+    """
+    Compute a correlation matrix over all datasets together. Data will be joined using the specified key.
+    TODO: What if different datasets use different keys to mean the same thing? How do we specify this?
+    """
+    results = _dispatch_tasks(client, data, *args, method='get_data', **kwargs)
```

Comment:
Yeah, I think it would be nice to factor out the data merging part. You might also need it for other functions.

Reply:
Created a separate function